.github/workflows/checkstyle.yaml

```diff
@@ -13,32 +13,39 @@ jobs:
         ref: ${{ github.event.pull_request.head.sha }}
     - name: Install dependencies
       run: |
+        echo 'force-unsafe-io' | sudo tee /etc/dpkg/dpkg.cfg.d/force_unsafe_io
         sudo apt-get update
-        sudo apt-get install --yes -qq build-essential autoconf libtool gawk alien fakeroot linux-headers-$(uname -r)
-        sudo apt-get install --yes -qq zlib1g-dev uuid-dev libattr1-dev libblkid-dev libselinux-dev libudev-dev libssl-dev python3 python3-dev python3-setuptools python3-cffi
-        # packages for tests
-        sudo apt-get install --yes -qq parted lsscsi ksh attr acl nfs-kernel-server fio
-        sudo apt-get install --yes -qq mandoc cppcheck pax-utils devscripts
-        sudo -E pip --quiet install flake8
+        xargs --arg-file=${{ github.workspace }}/.github/workflows/build-dependencies.txt sudo apt-get install -qq
+        sudo apt-get install -qq mandoc cppcheck pax-utils devscripts
+        sudo python3 -m pip install --quiet flake8~=4.0.1
```

Comment:
What's the purpose of pinning flake's version?

Reply:
Breaking changes in flake8 package (e. g. new checks added) would make workflows fail for changes completely unrelated to Python code. Pinning on major/minor prevents that as long as the developers there follow semantic versioning.

Reply:
> (e. g. new checks added) 

But it's great to have latest checks as fast as we can. Hard exclusions/blacklist/pinning lives long, unfortunately.

It's ok to pin if there is an exact problem, but it works pretty well without pinning and gives us fresh changes without manual bumps and tracking. 

My main thought is - at least I am ready to fix such problems when they arrive, and I propose not to harden CI stability, but use fresh code and be ready for problems. It's the same point as with latest distro release.

Reply:
I like your approach, even though it is quite unusual. People are usually bothered when all of the sudden workflows break.